### DISCUSSION TITLE
Events that trigger workflows at an org level

### DISCUSSION THREAD

**Discussion by heckerdj** [community/community#31500] [Enterprise]
I currently have a workflow that runs on a cron basis hitting our GitHub Enterprise server with API calls. This workflow is designed to ensure every repo in my organization has certain branch protection rules on their default branch. I would like to be able to have this workflow be event based instead of a cron job to limit the number of API calls it makes and reduce the load. Ideally I would like to trigger the workflow for every [branch_protection_rule](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#branch_protection_rule) event on any repo in the org. 

I looked into adding a webhook at the org level for the branch protection rules event but I cant seem to get it to trigger a workflow with the repository_dispatch on clause. Is this currently possible without creating an API endpoint to handle the event?
Is there a way I am missing in the documentation for the [Events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) that allows me to trigger at an org wide level? If not, is this a future planned feature?

**Comment by jmiller-rise8** [comment]
@heckerdj My org has created https://github.com/marketplace/webhook-actions to allow for this. Currently we can only offer a free plan until we reach 100 installations.

**Reply by heckerdj** [reply]
@jmiller-rise8 This is super cool, its exactly what I was looking at doing. I implemented a much less clean version internally to work for our GHE server for now. I would expect others to want this functionality too and thought we could drum up some conversation so that GitHub might implement and support it natively but it looks like your app is probably the way to go for now on github.com

**Reply by joshuaNathaniel** [reply]
@heckerdj 

Thank you! I appreciate the kind feedback. I have some other features in the backlog:
- a central config that can override repository configs giving admins (or a security team?) the ability to lockdown events
- a querying mechanism (something akin to mongodb queries) that operates on the webhook payloads essentially removing the need for conditionals in workflows

I realize you're using it on server, but if we were able to provide all of this to GHE Server would it be something that interests you?
